src/ripple/app/ledger/impl/LedgerReplayTask.cpp

```diff
@@ -116,6 +116,23 @@ void
 LedgerReplayTask::init()
 {
     JLOG(journal_.debug()) << "Task start " << hash_;
+
+    std::weak_ptr<LedgerReplayTask> wptr = shared_from_this();
+    skipListAcquirer_->addDataCallback(
+        [wptr](
+            bool good,
+            uint256 const& hash,
+            std::uint32_t seq,
+            std::shared_ptr<std::vector<uint256> const> const& skipList) {
+            if (auto sptr = wptr.lock(); sptr)
```

Comment:
In what situation would this check fail? Mainly, when would the `LedgerReplayTask` get destroyed before the `SkipListAcquire` calls it's callback?

Reply:
You are right. Currently no. Being defensive.

Reply:
Can we add an assert then? Since you don't think it should happen

Reply:
I realized that there could be a case during the `LedgerReplayer` `Stoppable` stops. The `LedgerReplayTask` could be destroyed right before the `if` statement. Sorry, I realized that when trying to add the assert...